### DISCUSSION TITLE
更新脚本无法更新uosc

### DISCUSSION THREAD

**Discussion by wli189** [dyphire/mpv-config#38] [General]
[  12.676][i][manager] updating https://github.com/dyphire/uosc... 
[  12.676][d][global] config path: 'scripts' -> 'D:/Program Files/mpv/portable_config/scripts'
[  12.677][d][global] user path: '~~/scripts' -> 'D:/Program Files/mpv/portable_config/scripts'
[  12.677][d][cplayer] Run command: subprocess, flags=64, args=[args="git,-C,D:/Program Files/mpv/portable_config/scripts,remote,add,manager,https://github.com/dyphire/uosc", playback_only="no", capture_size="67108864", capture_stdout="yes", capture_stderr="no", detach="no", env="", stdin_data="", passthrough_stdin="no"]
[  12.721][e][manager] error: remote manager already exists.
[  12.725][d][cplayer] Run command: subprocess, flags=64, args=[args="git,-C,D:/Program Files/mpv/portable_config/scripts,remote,set-url,manager,https://github.com/dyphire/uosc", playback_only="no", capture_size="67108864", capture_stdout="yes", capture_stderr="no", detach="no", env="", stdin_data="", passthrough_stdin="no"]
[  12.781][d][cplayer] Run command: subprocess, flags=64, args=[args="git,-C,D:/Program Files/mpv/portable_config/scripts,fetch,manager,dev", playback_only="no", capture_size="67108864", capture_stdout="yes", capture_stderr="no", detach="no", env="", stdin_data="", passthrough_stdin="no"]
[  13.426][e][manager] From https://github.com/dyphire/uosc
[  13.427][e][manager]  * branch            dev        -> FETCH_HEAD
[  13.427][e][manager]  + 564ebd0...7cd5fee dev        -> manager/dev  (forced update)
[  13.451][d][cplayer] Run command: subprocess, flags=64, args=[args="git,-C,D:/Program Files/mpv/portable_config/scripts,ls-tree,-r,--name-only,remotes/manager/dev", playback_only="no", capture_size="67108864", capture_stdout="yes", capture_stderr="no", detach="no", env="", stdin_data="", passthrough_stdin="no"]
[  13.512][e][manager] FAILED

**Comment by wli189** [comment]
manager. json里改为
  {
    "git":"https://github.com/dyphire/uosc",
    "whitelist":"^scripts/.*$",
    "dest":"~~/scripts/uosc",
    "branch":"dev"
  },

**Comment by dyphire** [comment]
感谢报错，将在上游此[pr](https://github.com/tomasklaen/uosc/pull/744)合并后推送相关修复

**Comment by dyphire** [comment]
由提交 5a49699 修复

**Comment by wli189** [comment]
而且https://github.com/Artoriuz/glsl-pixel-clipper
和https://github.com/Artoriuz/glsl-joint-bilateral
没有设定branch也是不可以的！更新默认下的是master分支，而这两个项目为main

**Comment by dyphire** [comment]
再次尝试修复： b4444d5
老实说一键更新这个功能我已经很久没用了。由于它创建一堆 .git 文件夹清理起来很不便，很长时间我都是手动更新一些值得关注的脚本着色器了
